FWCore/Framework/src/EventSetupRecordImpl.cc

```diff
@@ -27,31 +24,40 @@
 
 namespace edm {
   namespace eventsetup {
-    //
-    // constants, enums and typedefs
-    //
-
-    //
-    // static data member definitions
-    //
-
-    //
-    // constructors and destructor
-    //
-    EventSetupRecordImpl::EventSetupRecordImpl(EventSetupRecordKey const& iKey)
+
+    EventSetupRecordImpl::EventSetupRecordImpl(EventSetupRecordKey const& iKey,
+                                               ActivityRegistry const* activityRegistry,
+                                               unsigned int iovIndex)
         : validity_(),
           key_(iKey),
-          proxies_(),
-          eventSetup_(nullptr),
-          cacheIdentifier_(1),  //start with 1 since 0 means we haven't checked yet
-          transientAccessRequested_(false) {}
-
-    //
-    // member functions
-    //
+          activityRegistry_(activityRegistry),
+          cacheIdentifier_(1),
```

Comment:
Hmmm. This initial value is not actually used (at least it shouldn't be). It gets overwritten by the value from EventSetupRecordIOVQueue at the start of each IOV. I just initialized it here because it is not performance critical and initializing is "safer" practice and avoids things like compiler warnings. I'll add a comment that indicates this unless you suggest some better style for handling this situation.

Reply:
I pushed one commit that I hope resolves this and the next two comments. I combined the functions that set the validity interval and cache identifier into 1 function and added some comments that hopefully more clearly explain how this part of the code works.